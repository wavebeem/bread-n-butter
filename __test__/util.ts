import * as bnb from "../src/bread-n-butter";

export function snapTest<A>(parser: bnb.Parser<A>, input: string): void {
  expect(parser.parse(input)).toMatchSnapshot(JSON.stringify(input));
}
