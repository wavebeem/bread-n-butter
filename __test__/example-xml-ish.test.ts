import XML from "../examples/xml-ish";
import { snapTest } from "./util";

test("xml basic", () => {
  expect(XML.tryParse(`<a key="val" />`)).toEqual({
    name: "a",
    attributes: { key: "val" },
    children: [],
  });
  expect(XML.tryParse(`<a key="val">foo<b /></a>`)).toEqual({
    name: "a",
    attributes: { key: "val" },
    children: ["foo", { name: "b", attributes: {}, children: [] }],
  });
});

test("xml large example", () => {
  snapTest(
    XML,
    `
    <Map>
      <Entry key="string-a"><String>alpha</String></Entry>
      <Entry key="string-b"><String>bravo</String></Entry>
      <Entry key="string-c"><String>charlie</String></Entry>
      <Entry key="list">
        <List>
          <String>a</String>
          <String>b</String>
          <String>c</String>
        </List>
      </Entry>
      <Entry key="null"><Null /></Entry>
      <Entry key="empty" />
    </Map>
    `
  );
});
