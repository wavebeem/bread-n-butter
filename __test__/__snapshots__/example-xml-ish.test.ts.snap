// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`xml large example: "\\n    <Map>\\n      <Entry key=\\"string-a\\"><String>alpha</String></Entry>\\n      <Entry key=\\"string-b\\"><String>bravo</String></Entry>\\n      <Entry key=\\"string-c\\"><String>charlie</String></Entry>\\n      <Entry key=\\"list\\">\\n        <List>\\n          <String>a</String>\\n          <String>b</String>\\n          <String>c</String>\\n        </List>\\n      </Entry>\\n      <Entry key=\\"null\\"><Null /></Entry>\\n      <Entry key=\\"empty\\" />\\n    </Map>\\n    " 1`] = `
Object {
  "type": "ParseOK",
  "value": Object {
    "attributes": Object {},
    "children": Array [
      "
      ",
      Object {
        "attributes": Object {
          "key": "string-a",
        },
        "children": Array [
          Object {
            "attributes": Object {},
            "children": Array [
              "alpha",
            ],
            "name": "String",
          },
        ],
        "name": "Entry",
      },
      Object {
        "attributes": Object {
          "key": "string-b",
        },
        "children": Array [
          Object {
            "attributes": Object {},
            "children": Array [
              "bravo",
            ],
            "name": "String",
          },
        ],
        "name": "Entry",
      },
      Object {
        "attributes": Object {
          "key": "string-c",
        },
        "children": Array [
          Object {
            "attributes": Object {},
            "children": Array [
              "charlie",
            ],
            "name": "String",
          },
        ],
        "name": "Entry",
      },
      Object {
        "attributes": Object {
          "key": "list",
        },
        "children": Array [
          "
        ",
          Object {
            "attributes": Object {},
            "children": Array [
              "
          ",
              Object {
                "attributes": Object {},
                "children": Array [
                  "a",
                ],
                "name": "String",
              },
              Object {
                "attributes": Object {},
                "children": Array [
                  "b",
                ],
                "name": "String",
              },
              Object {
                "attributes": Object {},
                "children": Array [
                  "c",
                ],
                "name": "String",
              },
            ],
            "name": "List",
          },
        ],
        "name": "Entry",
      },
      Object {
        "attributes": Object {
          "key": "null",
        },
        "children": Array [
          Object {
            "attributes": Object {},
            "children": Array [],
            "name": "Null",
          },
        ],
        "name": "Entry",
      },
      Object {
        "attributes": Object {
          "key": "empty",
        },
        "children": Array [],
        "name": "Entry",
      },
    ],
    "name": "Map",
  },
}
`;
